react-components/src/context/UserContext.js

```diff
@@ -135,14 +135,7 @@ const UserContextProvider = props => {
     };
 
     const signOut = async () => {
-        try {
-            await revokeCustomerToken();
-            setCartCookie('', 0);
-            setUserCookie('', 0);
-            dispatch({ type: 'signOut' });
-        } catch (error) {
-            console.error('An error occurred during sign-out', error);
-        }
+        await signOutUserAction({ revokeCustomerToken, setCartCookie, setUserCookie, dispatch });
```

Comment:
The actual mutation should come from the consumer, i.e. the `signOut` function should receive the `revokeCustomerToken` as an argument (see the `resetCustomerCart` below, it uses the same principle).

The `getCustomerDetails` function is an anomaly, will fix it when we start working at the whole "My Account" stuf.